internal/commands/vaultsecrets/secrets/create.go

```diff
@@ -146,6 +146,8 @@ type MongoDBScope struct {
 var (
 	TwilioKeys = []string{"rotation_policy_name"}
 	MongoKeys  = []string{"rotation_policy_name", "mongodb_group_id"}
+	AwsKeys    = []string{"default_ttl", "role_arn"}
+	GcpKeys    = []string{"default_ttl", "service_account_email"}
```

Comment:
Happen to have an example of the YAML structure for say AWS? Trying to think if it makes sense to group related fields here similar to how we do in the API. 

For example, the AWS dynamic secret create API has a request body like:
```json
{
  "name": "s3_reader",
  "integration_name": "aws-integration",
  "default_ttl": "3600s",
  "assume_role": {
    "role_arn": "arn:aws:iam::644381835574:role/hvs-aws-s3-lister-prde"
  }
}
```

The `assume_role` field groups related fields. In the future, we might have an `iam_user` grouping for a different credential type.

Reply:
Yep, it would look something like [this](https://github.com/hashicorp/hcp/pull/143/files#diff-f00de939fa49d6f60296361e26cfa1dbca8436801089617b40f173ca64865310R222-R226). That's a great point, I think grouping fields (like `role_arn` in `assume_role`) would be great for organizing and even unmarshaling, but I wonder if it feels a bit unnecessary while there are so few fields atp? 

```
version: 1.0.0
type: "aws"
rotation_integration_name: "Aws-Int-12"
details: 
  default_ttl: "30"
  assume_role:
    role_arn: "ra"
```

....Then again, we would probably would want to introduce that structure now and build on that rather than wait til late when additional fields are added.

Reply:
Thought about it a bit more. I prefer your suggestion! Made [the change](https://github.com/hashicorp/hcp/pull/143/commits/12b7a5d75239dbb8691aea393e47685e2a8cbd51)

Reply:
Thanks!